bots/mirror/src/test/java/org/openjdk/skara/bots/mirror/MirrorBotFactoryTest.java

```diff
@@ -74,24 +76,203 @@ public void testCreate() {
+            String jsonString = """
+                    {
+                      "repositories": [
+                        {
+                          "from": "from1",
+                          "to": "to1",
+                          "branches": "master",
+                          "tags": "include"
+                        }
+                      ]
+                    }
+                    """;
+            var jsonConfig = JWCC.parse(jsonString).asObject();
+
+            var testBotFactory = TestBotFactory.newBuilder()
+                    .addHostedRepository("from1", new TestHostedRepository("from1"))
+                    .addHostedRepository("to1", new TestHostedRepository("to1"))
+                    .storagePath(tempFolder.path().resolve("storage"))
+                    .build();
+
+            assertThrows(IllegalStateException.class, () -> testBotFactory.createBots(MirrorBotFactory.NAME, jsonConfig));
+        }
+    }
+
+    @Test
+    public void testPassesWithBranchesAndTagsExplicitlyExcluded() {
+        try (var tempFolder = new TemporaryDirectory()) {
+            String jsonString = """
+                    {
+                      "repositories": [
+                        {
+                          "from": "from1",
+                          "to": "to1",
+                          "branches": "master",
+                          "tags": "exclude"
+                        }
+                      ]
+                    }
+                    """;
+            var jsonConfig = JWCC.parse(jsonString).asObject();
+
+            var testBotFactory = TestBotFactory.newBuilder()
+                    .addHostedRepository("from1", new TestHostedRepository("from1"))
+                    .addHostedRepository("to1", new TestHostedRepository("to1"))
+                    .storagePath(tempFolder.path().resolve("storage"))
+                    .build();
+
+            var bots = testBotFactory.createBots(MirrorBotFactory.NAME, jsonConfig);
+            assertEquals(1, bots.size());
+        }
+    }
+
+    @Test
+    public void testCreateWithTags() {
+        try (var tempFolder = new TemporaryDirectory()) {
+            String jsonString = """
+                    {
+                      "repositories": [
+                        {
+                          "from": "from1",
+                          "to": "to1",
+                          "branches": "master"
+                        },
+                        {
+                          "from": "from2",
+                          "to": "to2",
+                          "tags": "include"
+                        },
+                        {
+                          "from": "from3",
+                          "to": "to3",
+                          "tags": "only"
+                        },
+                        {
+                          "from": "from4",
+                          "to": "to4",
+                          "tags": "exclude"
+                        },
```

Comment:
I am wondering what's the expected behavior for the mirror bot configured like this? Intuitively speaking, I think the bot should mirror every branch but excluding tags. But currently, the bot would not mirror anything. if i am wrong please correct me. Thank you!

Reply:
I believe you are correct, this config will mirror nothing, but I agree that the expectation would be to mirror all branches but no tags.

Reply:
Yep, realized this after I pushed. I decided to remove `"exclude"` in the latest commit since implementing it is quite cumbersome (the `git` CLI client doesn't have a `--no-tags` option for `git push`).